File: legacy/tests.php5/testClosureCache.php
<?php

include_once ("java/Java.inc");

$here=getcwd();
java_require("$here/testClosureCache.jar");
function toString() {return "ccl";}
$tc = new Java("TestClosureCache");
"ichild::ccl"==$tc->proc1(java_closure(null, null, java('TestClosureCache$IChild'))) ||die(1);
"iface::ccl"==$tc->proc1(java_closure(null, null, java('TestClosureCache$IFace'))) || die(2);
echo "test okay\n";
?>

File: legacy/tests.php5/TestClosureCache.java
public class TestClosureCache {
    public interface IFace {}
    public interface IChild extends IFace {}
    public static String proc1(IChild o) { return "ichild::"+o;}
    public static String proc1(Object o) { return "object::"+o;}
    public static String proc1(IFace o) { return "iface::"+o;}
}
